### DISCUSSION TITLE
DBS Formulation

### DISCUSSION THREAD

**Discussion by POLonge** [idaholab/moose#28016] [Q&A Modules: Navier-Stokes]
### Check these boxes if you have followed the posting rules.

- [X] Q&A Navier Stokes is the most appropriate category for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (avoid screenshots if possible, triple back quotes before/after pasted text, etc)

### Question

Dear MOOSE Contributors @cpgr 

I'm seeking guidance and assistance in developing a MOOSE program (DBSFLOW) tailored for simulating reactive transport at the micro-continuum scale, specifically using the Darcy Brinkman formulation.

My objective is to model the complex interplay of fluid flow and reactive transport phenomena within porous media, focusing on the intricate details at the microscale. My proposed program would used more of the kernel developed in Navier Stoke and PorousFlow Modules with certain modification. The The Darcy Brinkman formulation is a sig=ngle pressure field equation that couple Navier stoke and Darcy into one equation. In the matrix the the reduces to darcy, as you go from porous region to solid region such as in vugs or fracture,  the equation asymptotically reduces to Navier stoke flow. 

I have attached below a figure of my proposed media, and research paper of this implementation in OPENFOAM.  II like to replicate this in MOOSE with improvement to capture slightly compressible flow that is applicable to CO2 storage in fractured-vuggy carbonate reservoir. I believe that MOOSE's capabilities are well-suited for this task, particularly with its flexibility and robustness in handling multiphysics problems.

[Multiphase flow modeling in multiscale porous media.pdf](https://github.com/user-attachments/files/16034566/Multiphase.flow.modeling.in.multiscale.porous.media.pdf)

![image](https://github.com/idaholab/moose/assets/146491923/98cdc962-f5b6-46e2-a21f-12e83198c8e5)

![image](https://github.com/idaholab/moose/assets/146491923/379dd659-3e40-4715-8e48-a5fc5d2c0a97)

Any specific guidance on how to implement the Darcy Brinkman formulation within the MOOSE framework? Additionally, I would appreciate any insights or best practices that could streamline the development process and ensure accurate simulation results.

**Comment by GiudGiud** [comment]
Hello

The three equations can be implemented in moose. Would the variables be velocity, phi and pressure?
Once you have identified the variables, you should decide which discretization to use. Both modules have a continuous galerkin finite element and a finite volume implementation that can serve as examples. 
The paper mentions two flow regimes can be present, Darcy and NS flow. Do you plan to have the same equation over the entire domain and the flow regime naturally lets some terms be important and others be negligible ?

We have tutorials on creating new applications here:
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/index.html

Chris will have more targeted advice

Guillaume
